dotnet/src/SemanticKernel/Memory/Collections/ScoredValue.cs

```diff
@@ -18,11 +18,18 @@ public ScoredValue(T item, double score)
     }
 
     public T Value { get; }
-    public Score Score { get; }
+    public double Score { get; }
 
     public int CompareTo(ScoredValue<T> other)
     {
-        return this.Score.CompareTo(other.Score);
+        var result = this.Score.CompareTo(other.Score);
+
+        if (result == 0 && this.Value is IComparable<T> comparableValue)
+        {
+            result = comparableValue.CompareTo(other.Value);
+        }
+
```

Comment:
We're making some statements here about how to "break a tie" by comparing score and value... a clever idea. That said, if I was checking for whether two entries are equal, I think I'd want to know that they are equal Score -- not sure I'd expect them to be differently ordered even when their Scores are identical. It's a design choice, and one that should be well-documented.